### PR TITLE
Support Jenkins 2.361 and migrate basic integration tests to pytest operator

### PR DESCRIPTION
**Pull request by jdkandersson**
- Changes to install the default jre version for all Ubuntu series
- Migrate the basic integration tests to pytest operator

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: lib/charms/layer/jenkins/packages.py [hunk 1]
```diff
@@ -130,20 +127,18 @@ def _setup_source(self, release):
...
     def jenkins_upgradable(self):
         """
-            Verify if there's a new version of jenkins available.
-            Note: When bundle-site is not set the return will always
-            be True.
+        Verify if there's a new version of jenkins available.
+        Note: When bundle-site is not set the return will always
+        be True.
         """
         if hookenv.config()["bundle-site"] == "":
```
Comment (arturo-seijas) [inline lib/charms/layer/jenkins/packages.py:139]: Picky observation: I'd go with merging both `if` and directly returning the value.

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+@pytest.mark.abort_on_fail
+async def test_tools_installed(app: Application):
+    """
+    arrange: given charm has been deployed and is idle
+    act: when checking for the python3-testtools package
+    assert: then it is found
+    """
+    dpkg_output = await app.units[0].ssh("dpkg -l python3-testtools")
+
+    assert "ii" in dpkg_output, "No tool installation found"
```
Comment (arturo-seijas) [inline tests/integration/test_charm.py:53]: ii?

File: lib/charms/layer/jenkins/packages.py (hunk 1 above)
Comment (jdkandersson) [reply lib/charms/layer/jenkins/packages.py:139]: Yeh there is a lot of clean up potential, this part of the code was only touched by black in this case. I would suggest changes like this one are out of scope for this PR

File: tests/integration/test_charm.py (hunk 2 above)
Comment (jdkandersson) [reply tests/integration/test_charm.py:53]: `ii`` appears in the `dpkg -l <package name>` output only if the package `<package name>` is installed. I have added a comment to clarify this

File: lib/charms/layer/jenkins/packages.py (hunk 1 above)
Comment (arturo-seijas) [reply lib/charms/layer/jenkins/packages.py:137]: Works for me

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Pull request merged**